padl/transforms.py

```diff
@@ -758,80 +788,52 @@ def pd_device(self) -> str:
     def pd_preprocess(self) -> "Transform":
         """The preprocessing part of the transform. The device must be propagated from self."""
         pre = self._pd_get_stages()[0]
-        pre.pd_to(self.pd_device)
+        pre.pd_set_device(self.pd_device)
         return pre
 
     @property
     def pd_forward(self) -> "Transform":
         """The forward part of the transform (that what's typically done on the GPU).
         The device must be propagated from self."""
         forward = self._pd_get_stages()[1]
-        forward.pd_to(self.pd_device)
+        forward.pd_set_device(self.pd_device)
```

Comment:
@wuhu I replaced `pd_to` here with `pd_set_device` because that was the original intention of adding `pd_to` here. It was to make sure that the splits inherited the correct device from the parent transform. However, sending the layers to device multiple times creates unnecessary overhead hence the change.

Reply:
:+1: